static/app/components/actions/actionLink.tsx

```diff
@@ -35,18 +51,33 @@ export default function ActionLink({
   confirmPriority,
   ...props
 }: Props) {
-  const action = (
-    <StyledAction
-      as={type === 'button' ? ActionButton : 'a'}
-      aria-label={title}
-      className={classNames(className, {disabled})}
-      onClick={disabled ? undefined : onAction}
-      disabled={disabled}
-      {...props}
-    >
-      {children}
-    </StyledAction>
-  );
+  let action = <React.Fragment />;
+
+  if (type === 'button') {
+    action = (
+      <StyledActionButton
+        aria-label={title}
+        className={classNames(className, {disabled})}
+        onClick={disabled ? undefined : onAction}
+        disabled={disabled}
+        {...props}
+      >
+        {children}
+      </StyledActionButton>
+    );
+  } else {
+    action = (
+      <StyledAction
+        aria-label={title}
+        className={classNames(className, {disabled})}
+        onClick={disabled ? undefined : onAction}
+        disabled={disabled}
+        {...props}
+      >
+        {children}
+      </StyledAction>
+    );
+  }
```

Comment:
```suggestion
  const actionCommonProps = {
    ['aria-label']: title,
    className: classNames(className, {disabled}),
    onClick: disabled ? undefined : onAction,
    disabled,
    children,
    ...props,
  };

  const action =
    type === 'button' ? (
      <StyledActionButton {...actionCommonProps} />
    ) : (
      <StyledAction {...actionCommonProps} />
    );
```

Reply:
Yeah this might work, let me try. I think I might have actually tried doing this

Reply:
That works!